agent/config/builder.go

```diff
@@ -1606,6 +1622,12 @@ func (b *Builder) tlsCipherSuites(name string, v *string) []uint16 {
 
 func (b *Builder) nodeName(v *string) string {
 	nodeName := b.stringVal(v)
+
+	var validDNSre = regexp.MustCompile(`^[a-zA-Z0-9]([a-zA-Z0-9\-]{0,62}[a-zA-Z0-9])?$`)
```

Comment:
Do you think it would be possible to use a single regex for both the invalid and valid cases? I notice this regex is a lot more strict than the InvalidDNS regex, but I'm not sure of the implications of these being different.